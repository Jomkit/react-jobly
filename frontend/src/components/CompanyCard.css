.CompanyCard-card:hover {
    color: blue;
}