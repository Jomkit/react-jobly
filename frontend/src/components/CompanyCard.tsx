import { Card, CardBody, CardText, CardTitle } from 'reactstrap';

interface companyPropsInterface {
    title: string;
    description: string;
    logoUrl?: string;
}

const CompanyCard = ({title, description, logoUrl}: companyPropsInterface) => {
  return (
    <li>
        <Card>
            <CardTitle tag="h3">
                {title}
            </CardTitle>
            <CardBody>
                <CardText>{description}</CardText>
                {logoUrl ? 
                <img
                    alt="company logo"
                    src={logoUrl} /> :
                <i className="fa-solid fa-image-portrait" style={{ fontSize: '100px' }}></i>
                }
            </CardBody>
        </Card>
    </li>
  )
}

export default CompanyCard