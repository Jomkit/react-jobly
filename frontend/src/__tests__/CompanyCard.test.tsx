import { render } from "@testing-library/react";
import CompanyCard from "../Features/Company/CompanyCard";
import { MemoryRouter } from "react-router-dom";
import { FAKEDATA } from "./_testCommon";

it('should render without crashing', () => { 
    let { handle, name, description, logoUrl } = FAKEDATA[0];
    render(
        <MemoryRouter>
            <CompanyCard key={handle} title={name} description={description} logoUrl={logoUrl} />
        </MemoryRouter>
    );
})

it('should match snapshot', () => { 
    let { handle, name, description, logoUrl } = FAKEDATA[0];
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard key={handle} title={name} description={description} logoUrl={logoUrl} />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
})