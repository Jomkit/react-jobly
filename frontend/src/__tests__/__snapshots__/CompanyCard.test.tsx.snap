// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <li>
    <div
      class="card"
    >
      <h3
        class="card-title"
      >
        test company 1
      </h3>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          this is a test
        </p>
        <img
          alt="company logo"
          src="test logo"
        />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
